### ISSUE TITLE
CRF status changes to "Data Entry Started" to early

### ISSUE THREAD

**Issue by oba-reliatec** [reliatec-gmbh/LibreClinica#376]
**Description:**
CRF status changes to "Data Entry Started" when CRF is touched, even if nothing is entered, changed or saved.

**Requirements:**
Scheduled event with an untouched CRF.

**Steps to follow:**
1. login
1. click on an previously untouched CRF in DataEntry mode (pencil icon)
1. leave CRF without doing anything
1. check the CRF's status icon

**Expected result:**
CRF status should remain "Not started".

**Actual result:**
CRF status is "Data Entry started".

This behaviour has uncomfortable consequences. For events with multiple optional CRFs, the user may open a CRF, but then realize that the CRF is not needed at all and leaves it unedited. As a result, the CRF is already created in the database. This cannot be undone. Unfortunately, this means that the event cannot be signed by the investigator because there is an "incomplete" CRF. Previously, it was possible to actually delete the CRF and thus fix the problem.
In one of the previous updates (probably already in OpenClinica), this was changed. Now only the data in the CRF will be deleted. The status of "Event CRF Started" remains on "available" and is no longer set back to "unavailable". This will block signing. Workaround is now to spoke the CRF "empty", with CRFs with mandatory fields this is a bit tedious.

**Comment by toskrip** [comment]
I don't think we should change this behaviour. I remember when it was changed in OC. Prior behaviour was that deleting of eCRF was actually deleting the item data and this was changed to better keep the audit trail (for GCP compliance) so that delete actually triggers item data modification with empty values.

For what you want to do you should not use delete but rather click on remove eCRF button. This will close all discrepancies on a form and change the status to invalid. Then you can sign the event.

**Comment by catrepte** [comment]
Unfortunately, the options "remove" is only available once the CRF has been saved at least once (see CRF CEC_Outcome in screenshot example). 
![CRF_with_no_data](https://github.com/reliatec-gmbh/LibreClinica/assets/133761401/45989940-bcec-4e8f-94c2-5be5499ece7d)

**Comment by catrepte** [comment]
I am a Data Manager and not so familiar with the backend of LibreClinica. In my experience, the situation occurs when data entry staff is unsure if they need to complete a CRF and just want to "take a look". To me it would not be necessary to know that they opened the CRF but saved no data (other than that they may need more training).

**Comment by toskrip** [comment]
If you have multiple data entry people working on project. It is good to display early that from is opened for data entry so that one person can see that college already started working on specific form (even more important if double data entry is configured). 

I understand the issues that you are describing, but the solution with not setting the EventCRF status early as described in ticket "expected result" I don't find good. The existing system documentation also describes this workflow that if CRF is opened inadvertently,  in order to change study the Event CRF needs to be deleted or removed. Which as you pointed is currently not possible on CRF without saved data. That is why new ticket should be opened that proposes changes to remove/delete CRF feature rather then changing the system behaviour of when the Data Entry Started status on CRF is set.

**Comment by oba-reliatec** [comment]
I agree to [catrepte](https://github.com/reliatec-gmbh/LibreClinica/issues/376#issuecomment-1549808812) that it is not necessary to know (also not in the audit trail from a GCP perspective) a user entered a CRF, did not save anything, and presumably left the CRF. As toskrip carefully remarked it might be an issue in double data entry with multiple users.
So how about to allow deleting an Event CRF with status data entry started given it was never saved from the database? This could be toggled by remove/delete CRF button consistent with the system documentation referred to by [toskrip](https://github.com/reliatec-gmbh/LibreClinica/issues/376#issuecomment-1550026822).

**Comment by toskrip** [comment]
It should be possible to implement toggle when the delete and remove button are displayed. To delete empty EventCRF we would need new implementation for deletion as the current one is just replacing the data from ItemData with empty values. So we would need a new implementation that deletes the EventCRF from table that would only be allowed when no item data exist. Then last thing to consider is the status of the event because that one changes as well when data entry starts on CRF. However I think we could leave that one be e.g. do not change event status when empty CRF from that event is deleted (could lead to situation where Event is data entry started event thou all CRFs are Not started.
